lib/types.d.ts

```diff
@@ -0,0 +1,151 @@
+import {IsEqual} from 'type-fest';
+
+/**
+ * Represents a merged object between `TObject` and `TOtherObject`.
+ * If both objects have a property with the same name, the type from `TOtherObject` is used.
+ */
+type MergeBy<TObject, TOtherObject> = Omit<TObject, keyof TOtherObject> & TOtherObject;
```

Comment:
can we use [`Merge`](https://github.com/sindresorhus/type-fest/blob/main/source/merge.d.ts) from type-fest?

Reply:
Done.